crates/stages/src/stages/bodies.rs

```diff
@@ -153,7 +156,11 @@ impl<DB: Database, D: BodyDownloader> Stage<DB> for BodyStage<D> {
             // Increment the transition if the block contains an addition block reward.
             // If the block does not have a reward, the transition will be the same as the
             // transition at the last transaction of this block.
-            let has_reward = self.consensus.has_block_reward(numhash.number());
+            let td = td_cursor
+                .seek(numhash)?
+                .ok_or(DatabaseIntegrityError::TotalDifficulty { number: numhash.number() })?
+                .1;
```

Comment:
Meh so on both Bodies + Execution we need to make 1 extra query to the TD stage for each block to make sure that we apply the correct block reward based on the TD.